### ISSUE TITLE
Issues in script

### ISSUE THREAD

**Issue by tosouravbhor** [itsToggle/plex_debrid#63]
Hi, and first of all, thank you so much for creating this fantastic project. The script I am trying to add to my plex server failed while I was trying to run it. I am a complete beginner with this project. I am done with mounting a real debris through rclone but still unable to get this script run. I am using Python3 latest, and the script gives me an error on line 91.  Could you please help me here, also rClone is not cloning properly on my server with the clone mount parameters you mentioned. I am. Having4 core, 8G RAM server with 200G SSD with a network port of 200 Mbps.

**Comment by itsToggle** [comment]
Hi, im sorry you are encountering these issues. Could you please tell me the exact error that the script is producing? line 91 looks fine to me. Make sure you have cloned this repository properly.

Same with rclone. What is the exact issue that you are encountering?

**Comment by tosouravbhor** [comment]
Hey I am extremely sorry. I am able to get the script working properly now. Seems like I didn't have proper packages installed and cloned script. Only one question if you guide me through which API to use with real debrid? I mean I am bit confused with lots of API available on Real-debrid API documentation. The issue with the rclone is that it is filling up my server every once in a while stating that I have no more space left.

**Comment by itsToggle** [comment]
You can define a maximum cache size for rclone by adding this tag to your mounting command: "--vfs-cache-max-size 10G". You can choose a different maximum size, this specific command will cap the maximum size at around 10GB.

What do you mean by different realdebrid APIs?

**Comment by tosouravbhor** [comment]
Hey thanks for the suggestion. I will update the mount command likewise as you suggested. I am at the stage of adding Real-debrid API after adding 'Plex' as a user. So I searched for Real-debrid API inside my account and found none. So I searched over the web and found this link, https://api.real-debrid.com/. Is it the same one? if yes then which one to use. Also, when it asks to add user, do I need to add this user in my plex home?

**Comment by itsToggle** [comment]
You need to enter your realdebrid api key, which you can find here: https://real-debrid.com/apitoken - its the same one that you had to enter when setting up rclone.

When adding a plex user, you can choose whatever name you want, the only important part is the plex token. How you find your Plex token is described when you set up the user inside plex_debrid :)

**Comment by tosouravbhor** [comment]
Hey thanks so much you helped me a lot with this issue and I am finally able to add media from my watchlist to the library automatically. Just one more time I am troubling you is about the transcoding of media. I mounted the drive using rclone and the parameters you gave me. But the only issue I am facing now is that the scanning is taking indefinite time. Am I doing something wrong here? I have set transcoding to ultra fast and prefer high speed encoding. Is there anything else that I need to do? Please note that I have disabled everything as per your suggestion on my server. Here is the screenshot of my server transcoding section, https://paste.pics/I4R3Y

**Comment by tosouravbhor** [comment]
Working so so fine... Man you are a legend. I would truly be a part of this amazing project and let me know how can I help you with. I am good with documentation so if required I can do something like that

**Comment by itsToggle** [comment]
Thank you ☺️ The actual scanning of the library is not related to your transcoding settings and should definitely be done in less than a minute.
 
Depending on your library settings though, Plex might be performing other operations in the background, which is why the "wheel" is still spinning. In the readme's Plex setup section I've written down some recommended settings for your libraries. These recommendations do not include disabling "Intro detection" for tv shows, which is what I'm guessing your server is doing at the moment. Intro detection takes a few minutes, depending on how many new episodes were added.

If the actual scanning of the library is taking a long time (and not, as I'm suggesting, the intro detection) I would need a bit more information on how big your library is and if you have any plex shares connected to your Plex account.

**Comment by tosouravbhor** [comment]
Hey thanks for another helpful suggestion and is there any way I can keep this service active in the background as well as opening it after server restart?

**Comment by itsToggle** [comment]
You can have Plex_debrid start in "run" mode directly by changing the setting "/Settings/UI settings/show menu on startup" to "false".

I'm sure you can find a lot of information on how to autostart and run a python script in the background on google :)

rclone can be run as a background service as detailed in the readme, and autostarting rclone is also just a quick google search :)

**Comment by tosouravbhor** [comment]
You are a legend!!!! You made my life way easier!!! Thanks a lot and I will surely go this way!!! Thanks a lot!!

**Comment by tosouravbhor** [comment]
Hey I have made the python script executable and set the cronjob like below, @reboot sleep 30 && /var/lib/transmission-daemon/test.py >/dev/null 2>&1 but it is not working after reboot. I need to run it manually and keep it running in the background. Sorry for this constant back and forth but I am new to this one. I am able to set rclone on cron though shell scripting but not this one. I have set the GUI flag to false but still no luck with cron job. Could you suggest what am I doing wrong here?

**Comment by itsToggle** [comment]
Hey, sorry I haven't set up an autostart routine for myself - so I won't be of any help. If you do figure it out, feel free to post your solution here or on the discussion section :)

**Comment by tosouravbhor** [comment]
Hey indeed I will. Thanks for all the support. Have a great week ahead.

**Comment by dspdhckr** [comment]
Well I tested thoroughly and found that I can start the utility when computer restarts by adding a cronjob with @reboot. This cronjob will open a screen (linux function) and start executing the script which I have already set to 'false' to open menu on startup. This way, without needing to keep terminal open, I am able to add movies to watchlist and watch them instantly. I am using Oracle free cloud.
